src/Exception/BatchException.php

```diff
@@ -2,61 +2,49 @@
 
 namespace Http\Client\Exception;
 
+use Http\Client\BatchResult;
+
 /**
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class BatchException extends TransferException
+final class BatchException extends RuntimeException
 {
     /**
-     * @var TransferException[]
+     * @var BatchResult
      */
-    private $exceptions;
+    private $result;
 
     /**
-     * @param TransferException[] $exceptions
+     * @param BatchResult $result
      */
-    public function __construct(array $exceptions = [])
+    public function __construct(BatchResult $result)
     {
-        parent::__construct('An error occurred when sending multiple requests.');
-
-        foreach ($exceptions as $e) {
-            if (!$e instanceof TransferException) {
-                throw new InvalidArgumentException('Exception is not an instanceof Http\Client\Exception\TransferException');
-            }
-        }
-
-        $this->exceptions = $exceptions;
+        $this->result = $result;
     }
 
     /**
-     * Returns all exceptions
+     * Returns successful responses or null
```

Comment:
Returns a list BatchResult which contains succesful responses and exceptions for failed requests.